apollo-adapters/src/appleMain/kotlin/com/apollographql/apollo3/adapter/BigDecimal.kt

```diff
@@ -8,9 +8,9 @@ actual class BigDecimal internal constructor(private val raw: NSDecimalNumber) :
 
   actual constructor(doubleVal: Double) : this(NSDecimalNumber(doubleVal))
 
-  actual constructor(intVal: Int) : this(NSDecimalNumber(intVal))
+  actual constructor(intVal: Int) : this(NSDecimalNumber(int = intVal))
 
-  actual constructor(longVal: Long) : this(NSDecimalNumber(long = longVal))
+  actual constructor(longVal: Long) : this(NSDecimalNumber(longLong = longVal))
```

Comment:
How is `longlong` better than `long`?

Reply:
Ahah that's a good question! On some targets, `NSDecimalNumber(long)` takes a `Long` whereas on others it takes an `Int`. However `NSDecimalNumber(longLong)` always takes a `Long`. Not sure if there was a better way to resolve this though!